docker/dockerfiles/dev-env-ubuntu.Dockerfile

```diff
@@ -0,0 +1,42 @@
+# This docker file build the Starrocks backend developement environment with all the tooling, dependencies libraries and
+# maven dependencies pre-installed.
+# Please run this command from the git repo root directory to build:
+#    DOCKER_BUILDKIT=1 docker build --rm=true -f docker/dockerfiles/dev-env-ubuntu.Dockerfile -t starrocks/dev-env-ubuntu:tag .
+
+FROM ubuntu:22.04 as base
+
+# Install common libraries and tools that are needed for dev environment
+RUN apt-get update -y && \
```

Comment:
going to add clang14 into dev env, which the compiling/link speed is much faster than gcc. :)